### PR TITLE
Port performance page to flutter

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Screen Shot 2019-12-02 at 8 00 23 PM
Screen Shot 2019-12-03 at 8 54 06 AM
Screen Shot 2019-12-03 at 8 54 13 AM

### CODE REVIEW

File: packages/devtools_app/lib/src/performance/flutter/performance_screen.dart [hunk 1]
```diff
@@ -29,82 +40,105 @@ class PerformanceScreen extends Screen {
...
+            _buildStateControls(),
+            ProfileGranularityDropdown(),
+          ],
+        ),
+        Expanded(
+          child: ValueListenableBuilder(
+            valueListenable: controller.cpuProfilerController.dataNotifier,
+            builder: (context, cpuProfileData, _) {
+              if (cpuProfileData ==
+                  controller.cpuProfilerController.dummyEmptyCpuProfileData) {
```
Comment (DaveShuckerow) [inline packages/devtools_app/lib/src/performance/flutter/performance_screen.dart]: Nit: if this object is a special one that indicates a recording is in progress, I suggest naming it to reflect that.

File: packages/devtools_app/lib/src/flutter/common_widgets.dart [hunk 2]
```diff
@@ -154,3 +156,92 @@ ErrorReporter showErrorSnackBar(BuildContext context) {
...
+  ];
+}
+
+Widget recordingInfo({
+  Key instructionsKey,
+  Key statusKey,
+  @required bool recording,
+  @required String recordedObject,
+}) {
+  final recordingInstructions = Column(
```
Comment (DaveShuckerow) [inline packages/devtools_app/lib/src/flutter/common_widgets.dart]: This code is very easy to read!

File: packages/devtools_app/lib/src/flutter/common_widgets.dart [hunk 3]
```diff
@@ -154,3 +156,92 @@ ErrorReporter showErrorSnackBar(BuildContext context) {
     );
   };
 }
+
+Widget clearButton({
```
Comment (DaveShuckerow) [inline packages/devtools_app/lib/src/flutter/common_widgets.dart]: Suggestion: make these StatelessWidgets, and add some documentation about how they're supposed to be used in different contexts.

File: packages/devtools_app/lib/src/flutter/common_widgets.dart [hunk 4]
```diff
@@ -154,3 +156,92 @@ ErrorReporter showErrorSnackBar(BuildContext context) {
...
+    onPressed: onPressed,
+    child: MaterialIconLabel(
+      Icons.block,
+      'Clear',
+      minIncludeTextWidth: minIncludeTextWidth,
+    ),
+  );
+}
+
+List<Widget> recordStopButtons({
```
Comment (DaveShuckerow) [inline packages/devtools_app/lib/src/flutter/common_widgets.dart]: Suggestion: 
Build the containing widget that presents these two buttons, or separate this list into two widgets: one for record and one for stop.

File: packages/devtools_app/lib/src/performance/flutter/performance_screen.dart [hunk 5]
```diff
@@ -29,82 +40,105 @@ class PerformanceScreen extends Screen {
...
+              minIncludeTextWidth: minIncludeTextWidth,
+              recording: recording,
+              onRecord: _startRecording,
+              onStop: _stopRecording,
+            ),
+            const SizedBox(width: 8.0),
+            clearButton(
+              key: PerformanceScreen.clearButtonKey,
+              minIncludeTextWidth: minIncludeTextWidth,
+              onPressed: recording
```
Comment (DaveShuckerow) [inline packages/devtools_app/lib/src/performance/flutter/performance_screen.dart]: The following will be the same:

```dart
onPressed: recording ? null : _clear,
```

`_clear` already is a `Future<void> Function()` so you can pass it as a tearoff.

File: packages/devtools_app/lib/src/profiler/cpu_profiler_controller.dart [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+import 'package:meta/meta.dart';
+
+import '../profiler/cpu_profile_model.dart';
+import '../profiler/cpu_profile_service.dart';
+import '../profiler/cpu_profile_transformer.dart';
+import '../ui/fake_flutter/fake_flutter.dart';
+
+class CpuProfilerController {
+  CpuProfilerController() {
+    dummyEmptyCpuProfileData = CpuProfileData.parse({});
```
Comment (DaveShuckerow) [inline packages/devtools_app/lib/src/profiler/cpu_profiler_controller.dart]: I don't see any other writes to either of these fields.  Can they be final instead?

File: packages/devtools_app/test/flutter/performance_screen_test.dart [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    testWidgets('builds its tab', (WidgetTester tester) async {
+      await tester.pumpWidget(wrap(Builder(builder: screen.buildTab)));
+      expect(find.text('Performance'), findsOneWidget);
+    });
+
+    const windowSize = Size(1000.0, 1000.0);
+
+    testWidgetsWithWindowSize(
```
Comment (DaveShuckerow) [inline packages/devtools_app/test/flutter/performance_screen_test.dart]: Out of curiosity, why is a customized windowSize needed here?

File: packages/devtools_app/lib/src/flutter/common_widgets.dart (hunk 3 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/flutter/common_widgets.dart]: done

File: packages/devtools_app/lib/src/flutter/common_widgets.dart (hunk 4 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/flutter/common_widgets.dart]: separated buttons and added documentation

File: packages/devtools_app/lib/src/performance/flutter/performance_screen.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/performance/flutter/performance_screen.dart]: renamed to baseStateCpuProfileData to better reflect its purpose

File: packages/devtools_app/lib/src/performance/flutter/performance_screen.dart (hunk 5 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/performance/flutter/performance_screen.dart]: done

File: packages/devtools_app/lib/src/profiler/cpu_profiler_controller.dart (hunk 6 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/profiler/cpu_profiler_controller.dart]: Done

File: packages/devtools_app/test/flutter/performance_screen_test.dart (hunk 7 above)
Comment (kenzieschmoll) [reply packages/devtools_app/test/flutter/performance_screen_test.dart]: The default creates too narrow of a window, and we get overflow errors

Comment (DaveShuckerow) [reply packages/devtools_app/test/flutter/performance_screen_test.dart]: File an issue & TODO to make this page work at narrow screen widths.  

For now, this is fine to submit.

**Pull request merged**